### PR TITLE
Improve POM

### PR DESCRIPTION
**Pull request by jponge**
Simple enhancements to the Maven POM

### CODE REVIEW

**Comment by katheris** [comment]
@jponge Thank you for the PRs! It's awesome to get your input. But before we merge your PRs we need you to accept the CLA. You can do this by just saying “I accept the Contributor License Agreement” as a comment, once you’ve read the CLA.md file in the top directory of the repo.

File: pom.xml [hunk 1]
```diff
@@ -6,51 +6,55 @@
 
   <groupId>kafka.vertx</groupId>
   <artifactId>demo</artifactId>
-  <version>0.0.1</version>
+  <version>0.0.2-SNAPSHOT</version>
 
   <properties>
     <kafka.version>2.3.0</kafka.version>
-    <vertx.version>3.8.4</vertx.version>
+    <vertx.version>4.0.0-SNAPSHOT</vertx.version>
```
Comment (katheris) [inline pom.xml]: This version of the vertx-dependencies doesn't seem to be in Maven central. Is there a snapshot repo we need to add to the pom.xml

File: src/main/java/kafka/vertx/demo/websocket/WebsocketVerticle.java [hunk 2]
```diff
@@ -25,7 +25,7 @@ public void start(Promise<Void> done) {
         HttpServer server = vertx.createHttpServer();
         Router router = Router.router(vertx);
         router.get().handler(StaticHandler.create());
-        server.websocketHandler(websocket -> {
+        server.webSocketHandler(websocket -> {
```
Comment (katheris) [inline src/main/java/kafka/vertx/demo/websocket/WebsocketVerticle.java]: In the latest milestone of vert.x 4.0.0 this function appears to still be spelt with a lowercase s

File: pom.xml (hunk 1 above)
Comment (jponge) [reply pom.xml]: We're using https://central.sonatype.org/pages/ossrh-guide.html to deploy snapshots, so you should add https://oss.sonatype.org/content/repositories/snapshots/ (preferrably in your `~/.m2/settings.xml`)

File: src/main/java/kafka/vertx/demo/websocket/WebsocketVerticle.java (hunk 2 above)
Comment (jponge) [reply src/main/java/kafka/vertx/demo/websocket/WebsocketVerticle.java]: This has changed in the current snapshots, and the latest milestone is back from December so currently fresh Vert.x 4 is in the snapshots.

**Comment by richardfontana** [comment]
@katheris Red Hat lawyer here. @jponge is a Red Hat employee. Just wanted to make sure that you want Red Hatters to agree to the CLA even though Red Hat is now a wholly owned subsidiary of IBM. Also incidentally I think the CLA has a error in it (it references the Eclipse Foundation, I'm assuming it's based on an old Eclipse CLA).

**Comment by katheris** [comment]
@richardfontana Hi, good spot on the CLA, that was my mistake I had an old copy of the recommended CLA. I have checked and we are now recommended to instead have contributors accept the DCO (https://github.com/hyperledger/fabric/blob/master/docs/source/DCO1.1.txt) I have add a CONTRIBUTING.md file to the repo. So the 4 parts of the CDO are what @jponge needs to confirm. If he just drops a message in here with his signature, with his consent I can add that to the commit when we squash and merge it. I also realised we had missed off the licences in the files as required by the Apache license so I've added them. We need everyone to accept the contributing guidelines just to sign off that it is all their own work. I have signed all my commits with "Signed-by: Katherine Stanley <katheris@uk.ibm.com>" to indicate this.

I hope that all makes sense and sorry for the confusion! We haven't had anyone accept me push code into this repo yet so this is the first time we've tested out the process!

**Comment by katheris** [comment]
The DCO:
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
660 York Street, Suite 102,
San Francisco, CA 94110 USA

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

**Comment by jponge** [comment]
I can certainly put some GnuPG-signed text here with the DCO text, but as @richardfontana pointed out I am already working for an IBM-owned entity, so is this really necessary? I am doing this in the context of work.

**Comment by katheris** [comment]
I'm afraid I've been told that it is, I'm an IBMer and still had to sign off my own work.

**Comment by jponge** [comment]
Ok @katheris, if @richardfontana is fine with the DCO text GPG-signed I will do that

**Comment by richardfontana** [comment]
@jponge Yes, we recommend the DCO for Red Hat-maintained projects too. I remember you wrote an [excellent blog post](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/) about the DCO a few years ago :) All that should be necessary is the "Signed-off-by: " statement.

**Comment by jponge** [comment]
Great thanks!

**Comment by jponge** [comment]
@katheris The last commits on all branches have been amended with sign-offs.

For the rest I accept the DCO:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512



Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

-----BEGIN PGP SIGNATURE-----
Version: Keybase OpenPGP v2.1.13
Comment: https://keybase.io/crypto

wsFcBAABCgAGBQJeb8VcAAoJEJlEcDP9Tmm0r7wQAJ1r+NT9PcTEUR0+7aYipYV9
0DU5RNw5HRM6Lt33e/RRlNQfk8vnCdzjmgDzK5cPDHpyS4exNu21LdqAMubqh9L0
26jYQ3dswzhaxtZs6I0KUmgL1vvZ+8PHMmlVkNXUUlHcjiurRSxs+8rG6av0G8A/
i+paBgC4IN05bjl18DeMobvIASCp8g7qB1Eay+hS0vpUcGEUm0H2U1VDNFaz53PP
RKc2DKLJbCplrrna78khYdqgObwc8qi2MF3O2u2B/cLqy29Yf6kyi9FoKzRzUA5o
KymPPuJ/4jaV6O1KWosN9SzAGvgfmox8SMpDo9GX4CcJzup7pBgHyhhPQRlbCHxf
uQ89fumVn2ALtPx8SFuHf2q8wtFs7nnQus2BmLT3SF36QrqScjrd2dZ+CwuXGWMl
PcPxxwwtuJ0zIJRLqpM89yemGX8o1xvz1IssMZWdC8E/8BaeExXLZB+WjLUnaGha
2Ylir/qbwdn4rTNtu4jvHLwOX3d47lDhkTNCrV9JtDQVk8mzPfehT2WhbzOYo8lQ
zyr8+xShE5gLo/RFwWpXpEvABWzIwTvz0gF93PCTvx4y81LUyd006uIzlZKzD7Jy
gNTwE09Z/lKmivGXOnVZUTRy9t/R5CxRdqeMMHaBYJZQjgWXBFujM+CwqSf7QVVs
swKhdX4m4NxVjAX38+du
=Utbb
-----END PGP SIGNATURE-----
```

(my key is available from Keybase at https://keybase.io/jponge)

**Comment by katheris** [comment]
Thank you @jponge I will get on and start merging some of your changes. And thank you again for your contributions, it's awesome and we appreciate it a lot!

**Pull request merged**